crates/modelardb_server/src/metadata/mod.rs

```diff
@@ -460,6 +499,244 @@ impl MetadataManager {
+    /// returned in sorted order by their start time. If no files belong to the column at
+    /// `query_schema_index` for the table with `model_table_name` an empty [`Vec`] is returned,
+    /// while an [`Error`] is returned if:
+    /// * The end time is before the start time.
+    /// * The max value is smaller than the min value.
+    /// * The metadata database could not be accessed.
+    /// * A model table with `model_table_name` does not exist.
+    pub async fn compressed_files_for_column_in_model_table(
+        &self,
+        model_table_name: &str,
+        query_schema_index: usize,
+        start_time: Option<Timestamp>,
+        end_time: Option<Timestamp>,
+        min_value: Option<Value>,
+        max_value: Option<Value>,
+    ) -> Result<Vec<Uuid>> {
+        // Set default values for the parts of the time and value range that is not defined.
+        let start_time = start_time.unwrap_or(0);
+        let end_time = end_time.unwrap_or(Timestamp::MAX);
+
+        if start_time > end_time {
+            return Err(Error::Configuration(Box::new(
+                ModelarDbError::DataRetrievalError(format!(
+                    "Start time '{start_time}' cannot be after end time '{end_time}'."
+                )),
+            )));
+        };
+
+        let min_value = min_value.unwrap_or(Value::NEG_INFINITY);
+        let max_value = max_value.unwrap_or(Value::INFINITY);
+
+        if min_value > max_value {
+            return Err(Error::Configuration(Box::new(
+                ModelarDbError::DataRetrievalError(format!(
+                    "Min value '{min_value}' cannot be larger than max value '{max_value}'."
+                )),
+            )));
+        };
+
+        let min_value = Self::rewrite_special_value_to_normal_value(min_value);
+        let max_value = Self::rewrite_special_value_to_normal_value(max_value);
+
+        let select_statement = format!(
+            "SELECT file_name FROM {model_table_name}_compressed_files
+             WHERE field_column = (?1)
+             AND (?2) <= end_time AND start_time <= (?3)
+             AND (?4) <= max_value AND min_value <= (?5)
+             ORDER BY start_time",
+        );
+
+        // query_schema_index is simply cast as a model table contains at most 1024 columns.
+        let mut rows = sqlx::query(select_statement.as_str())
+            .bind(query_schema_index as i64)
+            .bind(start_time)
+            .bind(end_time)
+            .bind(min_value)
+            .bind(max_value)
+            .fetch(&self.metadata_database_pool);
+
+        let mut file_names = vec![];
+        while let Some(row) = rows.try_next().await? {
+            file_names.push(row.try_get(0)?);
+        }
+
+        Ok(file_names)
+    }
+
+    /// Rewrite the special values [`Value`] (Negative Infinity, Infinity, and NaN) can represent to
+    /// the closets normal values in [`Value`] to simplify storing and querying them through SQLite.
```

Comment:
is this specific to SQLite or does it also benefit the other supported databases? If not general then maybe we should start to compile a list of what needs to change when moving to support more databases and add this to the list.

Reply:
It also benefits the other support RDBMSs, so I have changed `SQLite` to `metadata database`.